optuna/visualization/_intermediate_values.py

```diff
@@ -10,6 +14,35 @@
 _logger = get_logger(__name__)
 
 
+class _TrialInfo(NamedTuple):
+    trial_number: int
+    sorted_intermediate_values: List[Tuple[int, float]]
+
+
+class _IntermediatePlotInfo(NamedTuple):
+    trial_infos: List[_TrialInfo]
+
+
+def _get_intermediate_plot_info(study: Study) -> _IntermediatePlotInfo:
+    trials = study.get_trials(
+        deepcopy=False, states=(TrialState.PRUNED, TrialState.COMPLETE, TrialState.RUNNING)
+    )
+    trial_infos = [
+        _TrialInfo(trial.number, sorted(trial.intermediate_values.items()))
+        for trial in trials
+        if len(trial.intermediate_values) > 0
+    ]
+
+    if len(trials) == 0:
+        _logger.warning("Study instance does not contain trials.")
+    elif len(trial_infos) == 0:
+        _logger.warning(
+            "You need to set up the pruning feature to utilize `plot_intermediate_values()`"
```

Comment:
[nit] it would be great to update this comment because we can use this plot without the pruning feature (pruner class).